src/lem/circuit.rs

```diff
@@ -844,34 +1049,69 @@ impl Func {
                         // one constraint for the image's hash
                         num_constraints += 1;
                     }
-                    _ => todo!(),
+                    Op::Hide(..) => {
+                        // TODO
+                        globals.insert(FWrap(F::ZERO));
+                    }
+                    Op::Open(..) => {
+                        // TODO
+                        globals.insert(FWrap(F::ZERO));
+                    }
                 }
             }
             match &block.ctrl {
                 Ctrl::Return(vars) => num_constraints + 2 * vars.len(),
-                Ctrl::MatchTag(_, cases) => {
+                Ctrl::IfEq(_, _, eq_block, else_block) => {
+                    num_constraints
+                        + if nested { 6 } else { 4 }
```

Comment:
Why? I'm asking this because this function is also meant to be a high level documentation for the mechanics involved in `synthesize`

Reply:
Reason: `alloc_equal` consumes 4, `and` consumes 1 if nested, 0 if not nested